podpac/core/coordinates/coordinates.py

```diff
@@ -110,8 +114,20 @@ def __init__(self, coords, dims=None, coord_ref_sys=None, ctype=None, distance_u
             if isinstance(coords[i], BaseCoordinates):
                 c = coords[i]
             elif '_' in dim:
-                cs = [val if isinstance(val, Coordinates1d) else ArrayCoordinates1d(val) for val in coords[i]]
-                c = StackedCoordinates(cs)
+                # make sure components of stacked coordinates have dim names from stacked dim

```

Comment:
Hmmm. This is messy.

Is this a new check that we weren't doing before (to check for a name mismatch with existing Coordinates1d), or is this just because we need to set it now (instead of later) so we have to the check now?

I think you also need to check that `len(sdims) == len(coords[i])`.